### DISCUSSION TITLE
Best settings for large deletes

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#15366] [Q&A]
Is there an optimum way to do large delete operations. We are doing archiving of our DB and need to remove millions of old data.

Turn off all indexing and then do delete by query?

suggestions welcome.

**Comment by garayx** [comment]
yes, `DeleteByQueryOperation` should be good for this
https://ravendb.net/docs/article-page/5.1/Csharp/client-api/operations/delete-by-query

**Reply by bobskigit** [reply]
That takes forever we get max 200 to 300 docs/s.

**Reply by ayende** [reply]
Deletions are handled slowly, to avoid putting too much load on the system.
Is there any reason you cannot put this as a background task and just let it run?

**Reply by bobskigit** [reply]
I'm just concerned that it is many millions of items to delete. Is there an optimum set of items to delete in one go 1000, 10000, 1000000 or just go for the whole number?

**Reply by ayende** [reply]
No, let it run.
In general, you can also set the rate (docs / sec) for that 
Otherwise, it will operate in batches and take into account the overall load on the system.

Note that if this is an offline operation, you may want to _pause_ indexes, run this, then resume them.

**Reply by bobskigit** [reply]
Thanks for the advice, our cluster is very slow but your dev guys are looking at it.
